tests/Doctrine/Tests/ORM/Cache/Persister/Entity/AbstractEntityPersisterTest.php

```diff
@@ -258,23 +270,24 @@ public function testInvokeLoadById(): void
 
         $this->entityPersister->expects(self::once())
             ->method('loadById')
-            ->with(self::equalTo(['id' => 1]), self::equalTo($entity))
-            ->will(self::returnValue($entity));
+            ->with(self::identicalTo(['id' => 1]), self::identicalTo($entity))
+            ->willReturn($entity);
 
-        self::assertEquals($entity, $persister->loadById(['id' => 1], $entity));
+        self::assertSame($entity, $persister->loadById(['id' => 1], $entity));
     }
 
     public function testInvokeLoadOneToOneEntity(): void
     {
         $persister = $this->createPersisterDefault();
         $entity    = new Country('Foo');
+        $owner     = (object) [];
 
         $this->entityPersister->expects(self::once())
             ->method('loadOneToOneEntity')
-            ->with(self::equalTo([]), self::equalTo('foo'), self::equalTo(['id' => 11]))
-            ->will(self::returnValue($entity));
+            ->with(self::identicalTo([]), self::identicalTo($owner), self::identicalTo(['id' => 11]))
+            ->willReturn($entity);
 
-        self::assertEquals($entity, $persister->loadOneToOneEntity([], 'foo', ['id' => 11]));
```

Comment:
We're passing a string `'foo'` here although the method demands an object. As soon as we add native types, this is going to crash.